flatgfa/src/packedseq.rs

```diff
@@ -0,0 +1,293 @@
+use std::ops::Index;
+use std::fmt;
+static A: u8 = 0;
+static C: u8 = 1;
+static G: u8 = 2;
+static T: u8 = 3;
+
+/// A compressed vector-like structure for storing nucleotide sequences
+///     - Two base pairs are stored per byte
+///
+/// data: a vector that stores a compressed encoding of this vec's sequence
+/// high_nibble_end: whether the final base pair in the sequence is stored at a 
+///                   high or low nibble 
+struct PackedVec {
+    data: Vec<u8>,
+    high_nibble_end: bool,
+}
+
+impl PackedVec {
+    fn new() -> Self {
+        PackedVec {
+            data: Vec::new(),
+            high_nibble_end: false,
+        }
+    }
+    pub fn push(&mut self, value: u8) {
```

Comment:
Let's add an assertion here that `value` is within a safe range that it can be stored in a nibble. Maybe something like `assert!(value <= 0xF)`?